### PR TITLE
Use ellipsoid for coordinate calculations in KmlDataSource

### PR DESCRIPTION
**Pull request by tamarmot**
I've updated KmlDataSource to use an ellipsoid when calculating coordinates.  If no ellipsoid is passed in in the options, it defaults to WGS84.  

This addresses issue:
https://github.com/AnalyticalGraphicsInc/cesium/issues/6176

### CODE REVIEW

**Comment by cesium-concierge** [comment]
[Please sign the CLA before we review this PR.](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla)

Welcome to the Cesium community @tamarmot!

Can you please send in a [Contributor License Agreement](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) so that we can review and merge this pull request?

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by pjcozzi** [comment]
Thanks @tamarmot!  We received your CLA.

File: Source/DataSources/KmlDataSource.js [hunk 1]
```diff
@@ -1236,7 +1237,8 @@ define([
             oneTimeWarning('kml-gx:drawOrder', 'KML - gx:drawOrder is not supported in LineStrings');
         }
 
-        var coordinates = readCoordinates(coordinatesNode);
+        var ellipsoid = defaultValue(dataSource._ellipsoid, Ellipsoid.WGS84);
```
Comment (ggetz) [inline Source/DataSources/KmlDataSource.js]: Can we set this default value on the data source once before all these functions are called?

File: Source/DataSources/KmlDataSource.js [hunk 2]
```diff
@@ -2520,6 +2528,8 @@ define([
             up : defined(camera) ? Cartesian3.clone(camera.upWC) : undefined,
             bbox : defined(camera) ? camera.computeViewRectangle() : Rectangle.clone(Rectangle.MAX_VALUE)
         };
+
+        this._ellipsoid = options.ellipsoid;
```
Comment (ggetz) [inline Source/DataSources/KmlDataSource.js]: Yes, regarding my previous comment, right here use `defaultValue`, then use that value throughout.

File: Source/DataSources/KmlDataSource.js [hunk 3]
```diff
@@ -2468,6 +2475,7 @@ define([
      * @param {Object} options An object with the following properties:
      * @param {Camera} options.camera The camera that is used for viewRefreshModes and sending camera properties to network links.
      * @param {Canvas} options.canvas The canvas that is used for sending viewer properties to network links.
+     * @param {Ellipsoid} options.ellipsoid The global ellipsoid used for geographical calculations.  If undefined, WGS84 is used.
```
Comment (ggetz) [inline Source/DataSources/KmlDataSource.js]: `{Ellipsoid} options.ellipsoid The global ellipsoid used for geographical calculations.  If undefined, WGS84 is used.`

 Should be 

`{Ellipsoid} [options.ellipsoid=Ellipsoid.WGS84] The global ellipsoid used for geographical calculations.`

The `[]` are how you show an optional parameter, and we specify the default there.

**Comment by tamarmot** [comment]
Thanks Gabby,

Yes, I should write a spec and I have a sample test data that I can use for
inputs and expected outputs.  I will do so.

I initially had a default value going to this._ellipsoid but I did not have
access to the 'this' in the other functions, which was why I ended up
passing it in all over the place.

Thanks for the feedback, I'll review when I have a chance.  Do I assume you
want all the commits squashed before another pull request?

Tamar

**Comment by ggetz** [comment]
Thanks @tamarmot, no need to squash for now. Any commits you submit to your branch will be shown in this pull request.

**Comment by tamarmot** [comment]
Ok I wrote a couple of specs for testing coordinates in polygon and point.
They all end up calling the same function.
Thanks!

**Comment by ggetz** [comment]
Thanks @tamarmot, this looks good to me! Just make sure to mention the new option parameters (https://github.com/AnalyticalGraphicsInc/cesium/pull/6189/files) in `CHANGES.md`.

**Comment by tamarmot** [comment]
done, thanks

**Comment by hpinkos** [comment]
@ggetz is ready to go?  I looks good to me

**Comment by ggetz** [comment]
Yep! Thanks again @tamarmot !

**Pull request merged**